### PR TITLE
Feature/validation testing

### PR DESCRIPTION
**Pull request by sergio-usma**
### 📝 Description:
In this pull request, we have implemented the following project requirements in the past tense:

**✅ For the User model:**

- Ensured that "Name" is not blank.
- Verified that "PostsCounter" is an integer greater than or equal to zero.

**✅ For the Post model:**

- Ensured that "Title" is not blank.
- Limited "Title" to a maximum of 250 characters.
- Verified that "CommentsCounter" is an integer greater than or equal to zero.
- Ensured that "LikesCounter" is an integer greater than or equal to zero.

**✅ Additionally, unit specs have been added to cover all methods and validations for both User and Post models.**

🧪 Testing:

Unit tests have been written and successfully run to validate the implemented changes.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    post = Post.new(comments_counter: 0, likes_counter: 0, author_id: user.id, author: user)
+    expect(post).to_not be_valid
+  end
+
+  it 'is invalid with a negative comments_counter' do
+    user = User.create(name: 'John Doe')
+    post = Post.new(title: 'Lorem ipsum', comments_counter: -1, likes_counter: 0, author_id: user.id, author: user)
+    expect(post).to_not be_valid
+  end
+end
```
Comment (anagudelogu) [inline spec/models/post_spec.rb]: - [x] You did a great job working on this milestone! 💯 However, according to the requirements:

image

Please make sure to include tests for ALL your model's methods (`recent_comments`, `update_post_counter`) and all the validations: 

image

I can't see a test for the title's length and likes counter.

- [x] **OPTIONAL** When you're testing validations, is often a good practice to always test for both negative and positive cases, for example, if you're testing the `title's presence`, write a test to describe what happens if the title is missing and what happens if not. 🤓 🔧

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  it 'is invalid without a name' do
+    user = User.new(posts_counter: 0)
+    expect(user).to_not be_valid
+  end
+
+  it 'is invalid with a negative posts_counter' do
+    user = User.new(name: 'John Doe', posts_counter: -1)
+    expect(user).to_not be_valid
+  end
```
Comment (anagudelogu) [inline spec/models/user_spec.rb]: - [x] According to the requirements, please test all your model's methods. In this case, `#recent_posts` is missing its respective test. 🤓 🔧

File: spec/models/like_spec.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    like = Like.new(user_id: user.id)
+    expect(like).to_not be_valid
+  end
+
+  it 'is invalid without a user_id' do
+    user = User.create(name: 'John Doe')
+    post = Post.create(title: 'Lorem ipsum', comments_counter: 0, likes_counter: 0, author_id: user.id, author: user)
+    like = Like.new(post_id: post.id)
+    expect(like).to_not be_valid
+  end
```
Comment (anagudelogu) [inline spec/models/like_spec.rb]: - [x] According to the requirements, please test all your model's methods. In this case, `#update_post_counter` is missing its respective test. 🤓 🔧

File: spec/models/comment_spec.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    it 'belongs to a post' do
+      association = described_class.reflect_on_association(:post)
+      expect(association.macro).to eq :belongs_to
+    end
+
+    it 'belongs to a user' do
+      association = described_class.reflect_on_association(:user)
+      expect(association.macro).to eq :belongs_to
+    end
+  end
```
Comment (anagudelogu) [inline spec/models/comment_spec.rb]: - [x] According to the requirements, please test all your model's methods. In this case, `#update_post_counter` is missing its respective test. 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @sergio-usma, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb (hunk 1 above)
Comment (peterrobert) [reply spec/models/post_spec.rb]: This change has been done well

File: spec/models/user_spec.rb (hunk 2 above)
Comment (peterrobert) [reply spec/models/user_spec.rb]: This change has been done well

File: spec/models/like_spec.rb (hunk 3 above)
Comment (peterrobert) [reply spec/models/like_spec.rb]: This test has been added

File: spec/models/comment_spec.rb (hunk 4 above)
Comment (peterrobert) [reply spec/models/comment_spec.rb]: This test has been added

**Review by peterrobert — APPROVED** [review]
Hi @sergio-usma, 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️

- Great work making the changes requested by the previous reviewer. The test cases are also passing


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**